test/python/quantum_info/test_random.py

```diff
@@ -25,6 +26,23 @@
 class TestRandomUtils(QiskitTestCase):
     """Testing qiskit.quantum_info.random.utils"""
 
+    def test_seeded_random_unitary(self):
+        """Given the same seed, the output should be expected.
+        Following the numpy approach we are going to try to keep the policy of platform-independent
+        with randomness.
+        https://stackoverflow.com/questions/40676205/cross-platform-numpy-random-seed"""
+        seed = 314159
+        unitary = random_unitary(4, seed=seed)
```

Comment:
what's the point of creating a "random" unitary here, when this is always set to create the same thing?

Reply:
this file was created specifically to test this bug: https://github.com/Qiskit/qiskit-terra/issues/2850

seems like here you are using it to test the synthesis of a particular unitary (it is not random). for that, there are other tests. so i suggest just not adding this testcase.

Reply:
> what's the point of creating a "random" unitary here, when this is always set to create the same thing?

Tests the property of a fixed seed is creating a fixed output. If this test fails, means that the other tests (those setting the seed) are testing different cases.

> seems like here you are using it to test the synthesis of a particular unitary (it is not random). for that, there are other tests. so i suggest just not adding this testcase.

The description of the file matches this case. Do you have any specific suggestion on which file?

Reply:
So you are testing that setting seed produces the same random unitary regardless of platform? The link that you have provided essentially confirms this is the case, I don't know why Qiskit has to test for this. And if it fails some day, it is not a problem with Qiskit, but with numpy.

Reply:
Because we might accidentally break that property. Take the code from `two_qubit_decompose` as an example which we are both familiar with because recent issues.
The code `np.random.randn()*M2.real + np.random.randn()*M2.imag` might be change by `np.random.randn()*M2.imag + np.random.randn()*M2.real` under the assumption that they are both equivalent. However, `np.random.seed(3); np.random.randn()*M2.real + np.random.randn()*M2.imag` is not equal to `np.random.seed(3); np.random.randn()*M2.imag + np.random.randn()*M2.real`.

Reply:
ok good point.